server/socket/emitProjects/index.js

```diff
@@ -1,11 +1,53 @@
+import _ from 'lodash'
+import moment from 'moment'
 import { asyncRepeat } from 'common/utilities/asyncRepeat'
 import circleCI from 'common/circleCI'
 import { socket } from '../'
 
+const sortableTimeFormat = 'YYYYMMDDHHmmss'
+
+const getRecentlyBuiltBranches = branches => {
+  return _.chain(branches)
+    .map((branch, name) => ({ ...branch, name: name }))
+    .sortBy(branch => {
+      const recentBuilds = branch.recent_builds
+      return recentBuilds ? moment(recentBuilds[0].pushed_at).format(sortableTimeFormat) : 0
```

Comment:
what if `recentBuilds` is `[]`. Whenever you say: `something[0]` you have to think about the case that `something` is either `null` or `[]` - because it will break.

I suggest using `_.first` and checking that in your ternary instead.